### PR TITLE
In-code documentation for Unit, Result and Option types

### PR DESCRIPTION
**Pull request by pizycki**
This links to #9

### CODE REVIEW

File: Monacs.Core/Option.cs [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
         internal Option(T value)
         {
             Value = value;
             IsSome = true;
         }
-
+        
+        /// <summary>
+        /// Encapsulated valid value.
+        /// <para /> !!! It is not recommended to use it directly !!!
```
Comment (bartsokol) [inline Monacs.Core/Option.cs]: Avoid using exclamation marks, we don't have to scream at the consumers ;)

File: Monacs.Core/Option.cs [hunk 2]
```diff
@@ -1,21 +1,40 @@
...
+        /// </summary>
         public T Value { get; }
 
+        /// <summary>
+        /// Returns true if the option has a value that is not None.
+        /// </summary>
         public bool IsSome { get; }
 
+        /// <summary>
+        /// Returns true if the option has the None value.
```
Comment (bartsokol) [inline Monacs.Core/Option.cs]: `if the Option has no value` or `if the Option is None` sounds better

File: Monacs.Core/Option.cs [hunk 3]
```diff
@@ -1,21 +1,40 @@
...
+        /// <para /> Use one of the following extension methods instead:
+        /// <para /> * <see cref="Option.GetOrDefault{T}"/>, 
+        /// <para /> * <see cref="Option.Map{TIn,TOut}"/>, 
+        /// <para /> * <see cref="Option.Bind{TIn,TOut}"/>,
+        /// <para /> * <see cref="Option.Match{TIn,TOut}"/> 
+        /// </summary>
         public T Value { get; }
 
+        /// <summary>
+        /// Returns true if the option has a value that is not None.
```
Comment (bartsokol) [inline Monacs.Core/Option.cs]: `if the option has a value.` should be enough.

**Comment by bartsokol** [comment]
Please change the PR from `master` to `develop` branch.

File: Monacs.Core/Option.cs (hunk 1 above)
Comment (pizycki) [reply Monacs.Core/Option.cs]: @bartsokol we cannot bold or highlight and I wanted it to be never left unnoted:)
Do you think it should be formatted as normal text?

File: Monacs.Core/Option.cs (hunk 2 above)
Comment (pizycki) [reply Monacs.Core/Option.cs]: I will change to snd example

**Comment by pizycki** [comment]
Modified PR and changed some things.

**Comment by pizycki** [comment]
Actualy, I don't work on any part of this PR right now. The PR can be merged and I can take my hands on something else. Unless there is something to needed be fixed here.

**Pull request merged**